pkgs/by-name/ra/ratchet/package.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  callPackage,
+}:
+buildGoModule rec {
+  pname = "ratchet";
+  version = "0.9.2";
+
+  src = fetchFromGitHub {
+    owner = "sethvargo";
+    repo = "ratchet";
+    rev = "d57cc1a53c022d3f87c4820bc6b64384a06c8a07";
```

Comment:
Note that because the upstream package uses the variable passed to the linker in

```
"-X ${package_url}/internal/version.commit=${src.rev}"
```

here https://github.com/sethvargo/ratchet/blob/d57cc1a53c022d3f87c4820bc6b64384a06c8a07/internal/version/version.go#L16-L25

to construct the output for the `--version` flag, the nix package will produce different output from the release for `ratchet --version` if we do not hardcode and use the commit sha-1 here or elsewhere. This is in contrast to the more common convention in nixpkgs and used in the original PR to add ratchet to nixpkgs at 0.7.0

https://github.com/NixOS/nixpkgs/blob/70f6c4bb307cfd2b2081ff61f8eca4159cf79817/pkgs/by-name/ra/ratchet/package.nix#L11

Reply:
The other option would be to add `leaveDotGit = true;` to `src`, then get the commit hash for the specified tag using normal git commands.

This works too, however.

In either case, I would suggest adding a around `src` so that anyone updating the package or reviewing automated/proposed updates is aware of this and knows to look for the hash instead of the tag

Reply:
Many thanks for your review @RaghavSood . That sounds like a good idea. For now I will just mention the need to retain the hard-coded association between the version tag and its sha-1 head as the rev in a new comment. I could see an argument for transitioning to `leaveDotGit = true` and extracting the sha-1 with the git cli if you or anyone else feel there should be a strong preference for that.

Reply:
Here's another case where leaveDotGit is use for a similar purpose

https://github.com/NixOS/nixpkgs/pull/273438/files

It might be worth migrating to this approach in a future update.